x/e2ee/keyring/keyring.go

```diff
@@ -0,0 +1,202 @@
+package keyring
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/99designs/keyring"
+	"golang.org/x/crypto/bcrypt"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/client/input"
+	sdkkeyring "github.com/cosmos/cosmos-sdk/crypto/keyring"
+)
+
+const (
+	keyringFileDirName         = "e2ee-keyring-file"
+	keyringTestDirName         = "e2ee-keyring-test"
+	passKeyringPrefix          = "e2ee-keyring-%s"
+	maxPassphraseEntryAttempts = 3
+)
+
+type Keyring interface {
+	Get(string) ([]byte, error)
+	Set(string, []byte) error
+}
+
+func New(
+	appName, backend, rootDir string, userInput io.Reader,
+) (Keyring, error) {
+	var (
+		db  keyring.Keyring
+		err error
+	)
+	serviceName := appName + "-e2ee"
+	switch backend {
+	case sdkkeyring.BackendMemory:
+		return newKeystore(keyring.NewArrayKeyring(nil), sdkkeyring.BackendMemory), err
```

Comment:
maybe we can make use of sdkkeyring.New for the rest types

Reply:
the type is a private